src/dbsql.py

```diff
@@ -123,14 +129,34 @@ def _create_alert_api_call(self, query_id: str, alert: Alert, parent_folder: str
             rearm=alert.rearm,
         )
 
-    def _create_alert_schedule_api_call(self, alert: Alert, alert_id: str, destination_id: str, warehouse_id: str):
+    def _create_alert_schedule_api_call(self, alert: Alert, alert_id: str, destination_id: str, warehouse_id: str, run_as: str | None = None, tags: dict[str, str] = None):
         """
         Creates an alert schedule in Databricks
         """
+
+        if run_as:
+            if '@' in run_as:
+                run_as_obj = JobRunAs(user_name=run_as)
+            else:
+                run_as_obj = JobRunAs(service_principal_name=run_as)
+        else:
+            run_as_obj = None
+
+        if not tags:
+            tags = dict()
```

Comment:
Could  we avoid this check by initialising the `tags` arg in method call to `dict()`, instead of `None`?

Reply:
That is not recommended.
1) Mutable default values (https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments)
2) If you need to let the default value work, you have to either call the method twice with a condition or re-create the default values.

Reply:
Thinking about this. I am changing this to not to mutate the passed value.